### PR TITLE
Add sentry-native

### PR DESCRIPTION
**Pull request by SirLynix**
From https://github.com/getsentry/sentry-native

### CODE REVIEW

**Comment by SirLynix** [comment]
```
  Could NOT find CURL (missing: CURL_LIBRARY CURL_INCLUDE_DIR) (found version
  "7.73.0")
```

I'm beginning to think cmake package searching is just broken.

File: packages/s/sentry-native/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        add_syslinks("dbghelp")
+    elseif is_plat("linux") then
+        add_syslinks("dl", "pthread", "rt")
+    elseif is_plat("android") then
+        add_syslinks("dl", "log")
+    end
+
+    on_load("linux", "macosx", function (package)
+        package:add("deps", "libcurl")
+    end)
```
Comment (waruqi) [inline packages/s/sentry-native/xmake.lua:24]:
```
if is_plat("linux", "macosx") then
    add_deps("libcurl")
end
```

**Comment by waruqi** [comment]
>   "_CGColorGetTypeID", referenced from:
      CGColor* base::mac::CFCast<CGColor*>(void const* const&) in libmini_chromium.a(foundation_util.mm.o)
      CGColor* base::mac::CFCastStrict<CGColor*>(void const* const&) in libmini_chromium.a(foundation_util.mm.o)
  "_CTFontGetTypeID", referenced from:
      __CTFont const* base::mac::CFCast<__CTFont const*>(void const* const&) in libmini_chromium.a(foundation_util.mm.o)
      __CTFont const* base::mac::CFCastStrict<__CTFont const*>(void const* const&) in libmini_chromium.a(foundation_util.mm.o)
  "_CTRunGetTypeID", referenced from:
      __CTRun const* base::mac::CFCast<__CTRun const*>(void const* const&) in libmini_chromium.a(foundation_util.mm.o)
      __CTRun const* base::mac::CFCastStrict<__CTRun const*>(void const* const&) in libmini_chromium.a(foundation_util.mm.o)
  "_OBJC_CLASS_$_NSAutoreleasePool", referenced from:
      objc-class-ref in libmini_chromium.a(scoped_nsautorelease_pool.mm.o)
  "_OBJC_CLASS_$_NSFileManager", referenced from:
      objc-class-ref in libcrashpad_client.a(crash_report_database_mac.mm.o)
  "_audit_token_to_pid", referenced from:

We need use `add_frameworks()` to add some frameworks on macOS.

```
if is_plat("macosx") then
      add_frameworks("CoreGraphics", "CoreFoundation", "Foundation")
end
```

**Comment by SirLynix** [comment]
Will do that asap, kind of busy today

**Comment by SirLynix** [comment]
I have no idea why Windows errors like that. I have no trouble when I try to build it locally with cmake

File: packages/s/sentry-native/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            table.insert(configs, "-DBUILD_SHARED_LIBS=ON")
+            table.insert(configs, "-DSENTRY_BUILD_SHARED_LIBS=ON")
+        else
+            table.insert(configs, "-DBUILD_SHARED_LIBS=OFF")
+            table.insert(configs, "-DSENTRY_BUILD_SHARED_LIBS=OFF")
+        end
+        if package:is_plat("windows") then
+            opt.cxflags = { "/experimental:preprocessor-" } -- fixes <Windows SDK>\um\oaidl.h(487): error C2059: syntax error: '/'
+            local vs_runtime = package:config("vs_runtime")
+            table.insert(configs, "-SENTRY_BUILD_RUNTIMESTATIC=" .. ((vs_runtime == "MT" or vs_runtime == "MTd") and "ON" or "OFF"))
```
Comment (waruqi) [inline packages/s/sentry-native/xmake.lua:49]: `-SENTRY_BUILD_RUNTIMESTATIC`? `-DSENTRY_BUILD_RUNTIMESTATIC`?

maybe missing `-D`

**Comment by waruqi** [comment]
Maybe we need add `Winhttp`, `Shlwapi ` and `Advapi32 ` syslinks for windows

**Comment by SirLynix** [comment]
Seems good, thanks! 👍

**Pull request merged**